addons/stock/wizard/stock_picking_return.py

```diff
@@ -14,7 +14,7 @@ class ReturnPickingLine(models.TransientModel):
 
     product_id = fields.Many2one('product.product', string="Product", required=True, domain="[('id', '=', product_id)]")
     quantity = fields.Float("Quantity", digits=dp.get_precision('Product Unit of Measure'), required=True)
-    uom_id = fields.Many2one('uom.uom', string='Unit of Measure', related='move_id.product_uom', readonly=False)
+    uom_id = fields.Many2one('uom.uom', string='Unit of Measure', related='move_id.product_uom')
```

Comment:
```suggestion
    uom_id = fields.Many2one('uom.uom', string='Unit of Measure', related='product_id.uom_id')
```

Reply:
@ticodoo updated the related value and added a test in case of outgoing move's uom is different from product's uom